### PR TITLE
[PWGJE] jet v2 analysis

### PR DESCRIPTION
**Pull request by YubiaoWang**
analysis code for jet v2, include changed jet v2, in-plane and out-of-plane jet p_T distribution, \delta p_T distribution and several test plots

### CODE REVIEW

**Comment by raymondEhlers** [comment]
I don't have commit rights at the moment, so only took a quick look. My immediate question: by my count, you have 38 TH3Fs - have you checked on the memory usage?

**Comment by YubiaoWang** [comment]
> I don't have commit rights at the moment, so only took a quick look. My immediate question: by my count, you have 38 TH3Fs - have you checked on the memory usage?

Thanks for your comments. The memory usage is approximately 4GB when running the task at local test. Some of these TH3F plots are Jet QA plots, and they can be removed from the code if they take up too much memory.

**Comment by nzardosh** [comment]
> > I don't have commit rights at the moment, so only took a quick look. My immediate question: by my count, you have 38 TH3Fs - have you checked on the memory usage?
> 
> Thanks for your comments. The memory usage is approximately 4GB when running the task at local test. Some of these TH3F plots are Jet QA plots, and they can be removed from the code if they take up too much memory.

Hi,

4GB is way too large unfortunately. All the tasks together should be less than this overall

**Comment by YubiaoWang** [comment]
> > > I don't have commit rights at the moment, so only took a quick look. My immediate question: by my count, you have 38 TH3Fs - have you checked on the memory usage?
> > 
> > 
> > Thanks for your comments. The memory usage is approximately 4GB when running the task at local test. Some of these TH3F plots are Jet QA plots, and they can be removed from the code if they take up too much memory.
> 
> Hi,
> 
> 4GB is way too large unfortunately. All the tasks together should be less than this overall

**Comment by YubiaoWang** [comment]
> > > > I don't have commit rights at the moment, so only took a quick look. My immediate question: by my count, you have 38 TH3Fs - have you checked on the memory usage?
> > > 
> > > 
> > > Thanks for your comments. The memory usage is approximately 4GB when running the task at local test. Some of these TH3F plots are Jet QA plots, and they can be removed from the code if they take up too much memory.
> > 
> > 
> > Hi,
> > 4GB is way too large unfortunately. All the tasks together should be less than this overall

Thanks for your comments. I will remove some of these TH3F plots from the code and try use THnSparse.

**Review by nzardosh — DISMISSED** [review]

File: PWGJE/Tasks/jetchargedv2.cxx [hunk 1]
```diff
@@ -0,0 +1,786 @@
...
+  void fillRhoAreaSubtractedHistograms(T const& jet, float centrality, float rho, float weight = 1.0)
+  {
+    if (jet.r() == round(selectedJetsRadius * 100.0f)) {
+      registry.fill(HIST("h_jet_pt_rhoareasubtracted"), jet.pt() - (rho * jet.area()), weight);
+      registry.fill(HIST("h_jet_eta_rhoareasubtracted"), jet.eta(), weight);
+      registry.fill(HIST("h_jet_phi_rhoareasubtracted"), jet.phi(), weight);
+      registry.fill(HIST("h_jet_ntracks_rhoareasubtracted"), jet.tracksIds().size(), weight);
+      registry.fill(HIST("h2_centrality_jet_pt_rhoareasubtracted"), centrality, jet.pt() - (rho * jet.area()), weight);
+      if (jet.pt() - (rho * jet.area()) > 0) {
+        registry.fill(HIST("h2_centrality_jet_eta_rhoareasubtracted"), centrality, jet.eta(), weight);
```
[the remark this answers is not in the document]
Comment (YubiaoWang) [reply PWGJE/Tasks/jetchargedv2.cxx]: Thanks for your comment and sorry for not replying here in time, I remove those repeat plots in new version to make sure it doesn't take up a lot of memory

**Review by nzardosh — APPROVED** [review]

**Pull request merged**